web/core/components/core/charts/area-chart/root.tsx

```
@@ -0,0 +1,124 @@
+    yAxis,
+    className = "w-full h-96",
+    tickCount = {
+      x: undefined,
+      y: 10,
+    },
+    showTooltip = true,
+  } = props;
+  // derived values
+  const itemKeys = useMemo(() => areas.map((area) => area.key), [areas]);
+  const itemDotClassNames = useMemo(
+    () => areas.reduce((acc, area) => ({ ...acc, [area.key]: area.dotClassName }), {}),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)